src/sentry/api/event_search.py

```diff
@@ -157,7 +158,7 @@ def translate(pat):
 search_key           = key / quoted_key
 search_value         = quoted_value / value
 value                = ~r"[^()\s]*"
-numeric_value        = ~r"[-]?[0-9\.]+(?=\s|\)|$)"
+numeric_value        = ~r"[-]?[0-9\.]+[k|m]?(?=\s|\)|$)"
```

Comment:
Can we use regex groups (ie. wrap `([k|m|b]?)` in brackets) so that we don't have to use `.split` in the numeric parsing? This is what's happening for the `duration_format`